### DISCUSSION TITLE
What does sample_adj do?

### DISCUSSION THREAD

**Discussion by boxiangliu** [pyg-team/pytorch_geometric#2241] [Q&A]
Dear Matthias, 

Thank you for making pytorch-geometric. It has been a extremely useful tool. I am trying to use GraphSage in unsupervised training mode. To do that, I am following your graph_sage_unsup.py. As I am trying to understand the details of the `NeighborSampler`, I am lost on what `self.adj_t.sample_adj(n_id, size, replace=False)` (in sampler.py; line 152) is doing. 

Could you help me understand? Thank you in advance! 

Boxiang Liu, Ph.D.

**Comment by rusty1s** [comment]
`sample_adj` will sample `size` neighbors for each node in `n_id`. It's output should correspond to a sparse tensor of shape `[n_id.size(0), xxx]`, where `xxx` is determined by the number of unique neighbors sampled.

**Comment by eedalong** [comment]
@boxiangliu you can find code about sample_adj in [pytroch_sparse repo](https://github.com/rusty1s/pytorch_sparse/blob/master/torch_sparse/sample.py#L25)
